main.py

```diff
@@ -0,0 +1,49 @@
+import ssl
+from time import time
+from datetime import datetime
+from urllib.parse import urlparse
+from pathlib import Path
+import telebot
+
+# access token for telegram bot
+access_token = ''
+# the time to notify of the expiration of the certificate   
+notice_date = 432000
+bot = telebot.TeleBot(access_token)
+
+
+def check_certificate(url):
+    parsed_url = urlparse(url)
+    try:
+        cert = ssl.get_server_certificate(
+            (parsed_url.hostname, parsed_url.port or 443))
+    except Exception:
+        return "Bad request"
+    # save cert to temporary file (filename required for _test_decode_cert())
+    temp_filename = Path(__file__).parent / "temp.crt"
+    with open(temp_filename, "w") as f:
+        f.write(cert)
+    try:
+        parsed_cert = ssl._ssl._test_decode_cert(temp_filename)
+    except Exception:
+        return
```

Comment:
If your script will not be able to decode cert, it will be hard to troubleshoot what's going on because you handle but not return or log any exception